iOS/Sources/Beagle/Sources/Components/TabBar+Renderable/TabBarUIComponent.swift

```diff
@@ -17,136 +17,223 @@
     
     // MARK: - Model
     
     struct Model {
-        var tabIndex: Int
+        var tabIndex: Int?
         var tabBarItems: [TabBarItem]
-        var selectedTextColor: UIColor?
-        var unselectedTextColor: UIColor?
-        var selectedIconColor: UIColor?
-        var unselectedIconColor: UIColor?
+        var renderer: BeagleRenderer
     }
-
-    // MARK: - Properties
     
-    private var shouldScrollToCurrentTab = true
-    private var shouldAnimateOnCellDisplay = false
-    private var containerWidthConstraint: NSLayoutConstraint?
-    private let tabBarPreferedHeight: CGFloat = 65
-    private let collectionViewCellHeight: CGFloat = 55
+    // MARK: - Properties
+    private var shouldCreateTabItemsView = true
     private let tabItemMinimumHorizontalMargin: CGFloat = 40
     private let tabItemIconMinimunWidth: CGFloat = 75
     
     var model: Model
-    
+    var tabItemViews = [Int: TabBarItemUIComponent]()
     var onTabSelection: ((_ tab: Int) -> Void)?
-
+    
     // MARK: - UI
     
-    lazy var collectionView: UICollectionView = {
-        let layout = UICollectionViewFlowLayout()
-        layout.scrollDirection = .horizontal
-        layout.minimumInteritemSpacing = 0
-        layout.minimumLineSpacing = 0
-        let collection = UICollectionView(
-            frame: CGRect(),
-            collectionViewLayout: layout
-        )
-        collection.backgroundColor = .clear
-        collection.register(TabBarCollectionViewCell.self, forCellWithReuseIdentifier: TabBarCollectionViewCell.className)
-        collection.translatesAutoresizingMaskIntoConstraints = false
-        collection.showsHorizontalScrollIndicator = false
-        collection.dataSource = self
-        collection.delegate = self
-        return collection
+    private lazy var contentView: UIView = {
+        let view = UIView()
+        return view
     }()
     
-    lazy var containerIndicator: ContainerIndicatorView = {
-        let view = ContainerIndicatorView()
-        view.translatesAutoresizingMaskIntoConstraints = false
+    lazy var indicatorView: UIView = {
+        let view = UIView()
+        view.backgroundColor = .black
         return view
     }()
-    
+        
     // MARK: - Initialization
     
     init(
         model: Model
     ) {
         self.model = model
         super.init(frame: .zero)
-        setupLayout()
-        scrollTo(page: 0)
+        setupScrollView()
+        setupContentView()
     }
     
     @available(*, unavailable)
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
     
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        if let contentView = subviews.first {
+            contentSize = contentView.frame.size
+            resetTabItemsStyle()
+
+            // Creates tabItems only after view it's already in superview hierarchy
+            if superview != nil, shouldCreateTabItemsView {
```

Comment:
if let is more performatic to nil comparison.

```suggestion
            if let _ = superview, shouldCreateTabItemsView {
```